### ISSUE TITLE
ClientConfig type ACE

### ISSUE THREAD

**Issue by DetaliDigital** [modmore/ClientConfig#180]
Hello, Mark

When changing the type of the field from textarea to ace, the application simply stopped working. I show in the screenshot.

![изображение](https://user-images.githubusercontent.com/20732384/75099222-2f19cd00-55d0-11ea-81b5-01ace31569ae.png)

I show the ACE package of the latest version in the screenshot

![изображение](https://user-images.githubusercontent.com/20732384/75099269-a8192480-55d0-11ea-96d5-89469c1aa3af.png)

ACE Settings

![изображение](https://user-images.githubusercontent.com/20732384/75099340-8ec4a800-55d1-11ea-8c79-11f95c8431e7.png)

![изображение](https://user-images.githubusercontent.com/20732384/75099345-aac84980-55d1-11ea-9a5f-2139e069b5d6.png)

https://i.imgur.com/odhKlQv.png
![изображение](https://user-images.githubusercontent.com/20732384/75099371-f67af300-55d1-11ea-8c3e-0297ad4dbcf6.png)

### Environment
ClientConfig 2.3.0 , MODX 2.7.2 и PHP 7.2.25

**Comment by sebastian-marinescu** [comment]
Hi @OlegShchavelev just curious: did you fix it or find out what the problem was?

**Comment by Ruslan-Aleev** [comment]
Yep, I'm confirming bug. But problem in [Ace](https://github.com/modx-pro/modx-ace):

![focus_1](https://github.com/modmore/ClientConfig/assets/12523676/7480fb76-8b15-47c3-9bb5-3d570226b5bd)
![focus_2](https://github.com/modmore/ClientConfig/assets/12523676/73f80d84-2419-408d-92a6-282405a3af7d)

**Comment by sebastian-marinescu** [comment]
Hi @Ruslan-Aleev, what versions are you running? 

Because I **_cannot_** confirm it with either: 
`MODX 2.8.5-pl, Ace 1.9.3-pl, ClientConfig 2.3.3-pl`
nor with: 
`MODX 3.0.2-pl, Ace 1.9.3-pl, ClientConfig 2.3.0-pl`

Both are working for me: 

![image](https://github.com/modmore/ClientConfig/assets/3798871/d6f4a19f-1e24-413e-a719-d1eefb0f4c99)

**Comment by sebastian-marinescu** [comment]
Maybe it needs the system-setting to be also set to Ace?

![image](https://github.com/modmore/ClientConfig/assets/3798871/a662feee-56b6-41b7-bb7b-6a8ccd0b14e5)

**Comment by Ruslan-Aleev** [comment]
@sebastian-marinescu Hi!
It's really weird =)
I took up the question, since the problem was reported in the telegram chat, I checked the ClientConfig and the problem also happened to me.

I have: MODX 2.8.5-pl, Ace 1.9.3-pl (but Ace installed from modstore.pro), ClientConfig 2.3.3-pl

![image](https://github.com/modmore/ClientConfig/assets/12523676/e9c31c28-505b-4fb6-96ef-674d2a5a1f3f)

Not only that, I even found a solution =) but now it's not clear why you don't have a problem...

Solution:
in `/assets/components/ace/modx.texteditor.js` somewhere near the line 232, replace the code with:

```
focus: function (){
    this.valueHolder.focus();
},
```

**Comment by Ruslan-Aleev** [comment]
@sebastian-marinescu I installed clean MODX with git, did not do any MODX settings, I installed only 2 components Ace and ClientConfig from the modx.com repository.
I created a group and a code-setting in ClientConfig - the problem is reproduced...

**Comment by Ruslan-Aleev** [comment]
@sebastian-marinescu So, it seems that we found how to reproduce the bug - do the ace-setting **first in order, in group**. Then refresh the page and your ClientConfig section will fall with an error in the console.
